src/Client.ts

```diff
@@ -177,54 +181,172 @@ export class Client extends ClientJS {
           this.decorators.trigger(on.event, this, true)
         );
       } else {
-        this.on(
-          on.event as any,
-          this.decorators.trigger(on.event, this)
-        );
+        this.on(on.event as any, this.decorators.trigger(on.event, this));
       }
     });
 
-    return await super.login(token);
+    return super.login(token);
   }
 
   /**
    * Initialize all the @Slash with their permissions
    */
   async initSlashes() {
-    await Promise.all(
-      this.slashes.map(async (slash) => {
-        // Init all the @Slash
-        if (slash.guilds.length > 0) {
-          // If the @Slash is guild specific, add it to the guild
-          await Promise.all(
-            slash.guilds.map(async (guildID) => {
-              const guild = this.guilds.cache.get(guildID as Snowflake);
+    // # first init guild's all slash
+    const guildSlashStorage = new Map<string, DSlash[]>();
+    const guildsSlash = this.slashes.filter((s) => s.guilds?.length);
+
+    // group single guild slash togather
+    for (const s of guildsSlash) {
+      s.guilds.forEach((guild) =>
+        guildSlashStorage.set(guild, [
+          ...(guildSlashStorage.get(guild) ?? []),
+          s,
+        ])
+      );
+    }
```

Comment:
Could you explain what everything here from L200-207 does?

Reply:
1. it's collect how many guilds are used in slash.
2. group the guilds and its slash like ``guild > slash[]``
3. iterate over guilds and add/update/delete commands